pkg/common/resources/resources.go

```diff
@@ -751,6 +768,47 @@ func StrictlyGreaterThanOrEquals(larger, smaller *Resource) bool {
 	return true
 }
 
+// StrictlyGreaterThanOnlyExisting Return true if all quantities present or existing only in larger > smaller
+// Two resources that are equal are not considered strictly larger than each other.
+// Resource present in smaller but not in larger are not even considered.
+func StrictlyGreaterThanOnlyExisting(larger, smaller *Resource) bool {
+	if larger == nil {
+		larger = Zero
+	}
+	if smaller == nil {
+		smaller = Zero
+	}
+
+	// keep track of the number of not equal values
+	notEqual := false
+
+	// Is larger and smaller completely disjoint?
+	atleastOneResourcePresent := false
+	for k, v := range larger.Resources {
+		// even when smaller is empty, at least one of the resource type in larger should be greater than zero
+		if smaller.IsEmpty() && v > 0 {
+			return true
+		}
```

Comment:
if I have `{larger: {first: -10, second: 5}}` with an empty smaller this call will return `true`
That does not seem correct. It is not "strict" in that case.